src/zenml/event_sources/base_event_source.py

```diff
@@ -557,9 +561,12 @@ def get_event_source_config_schema(cls) -> Dict[str, Any]:
     def get_plugin_flavor_response_model(cls) -> EventFlavorResponse:
```

Comment:
I think you're missing the `hydrate` flag here here that tells you whether you populate the metadata or not.